src/core/simplify.rkt

```diff
@@ -76,11 +65,25 @@
 
      (egraph-func egg-graph node-ids iter-data))))
 
+(define (get-proof input start end)
+  (run-simplify-input
+    input
+    (lambda (egg-graph node-ids iter-data)
+      (define proof (egraph-get-proof egg-graph start end))
+      (when (null? proof)
+        (error (format "Failed to produce proof for ~a to ~a" start end)))
+      proof)))
 
 (module+ test
   (require "../syntax/types.rkt" "../syntax/rules.rkt")
-  (*needed-reprs* (list (get-representation 'binary64) (get-representation 'binary32)))
+  (define f64-repr (get-representation 'binary64))
+  (define f32-repr (get-representation 'binary32))
   (define all-simplify-rules (*simplify-rules*))
+  (define vars '(x a b c))
+
+  ;; set parameters
+  (*needed-reprs* (list f64-repr f32-repr))
+  (*context* (context vars f64-repr (map (const f64-repr) vars)))
```

Comment:
I think the `make-debug-context` function can be used here and avoid most of the diff.

Reply:
Didn't realize that was a thing.